tests/unit/conftest.py

```diff
@@ -197,25 +198,87 @@ def do_mock(
 
 @fixture
 def engine_id() -> str:
-    return "engine_id"
+    return "mock_engine_id"
 
 
 @fixture
-def get_engine_url(settings: Settings, account_id: str, engine_id: str) -> str:
+def engine_endpoint() -> str:
+    return "mock_engine_endpoint"
+
+
+@fixture
+def engine_name() -> str:
+    return "mock_engine_name"
+
+
+@fixture
+def get_engine_name_by_id_url(
+    settings: Settings, account_id: str, engine_id: str
+) -> str:
     return f"https://{settings.server}" + ACCOUNT_ENGINE_URL.format(
         account_id=account_id, engine_id=engine_id
     )
 
 
 @fixture
-def get_engine_callback(
-    get_engine_url: str, engine_id: str, settings: Settings
+def get_engine_id_by_name_url(
+    settings: Settings, account_id: str, engine_name: str
+) -> str:
+    return f"https://{settings.server}" + ACCOUNT_ENGINE_ID_BY_NAME_URL.format(
+        account_id=account_id
+    )
```

Comment:
You've can use the `engine_name` parameter here since the URL hit is actually `https://api.mock.firebolt.io/core/v1/accounts/mock_account_id/engines:getIdByName?engine_name=mock_engine_name`

e.g.
```
    return f"https://{settings.server}" + ACCOUNT_ENGINE_ID_BY_NAME_URL.format(
        account_id=account_id
    ) + f"?engine_name={engine_name}"
```

This way you don't need to do anything special in the actual test itself.

Reply:
Actually, it turns out that this and `get_engine_id_by_name_callback()` aren't being used. It's odd, because I expected them to be necessary in `test_connect_engine_name()`, but they're not. So I just deleted them.

Reply:
They're not used because you're doing it in the test itself, https://github.com/firebolt-db/firebolt-python-sdk/pull/195/files#diff-73d034cb3e24eab31a63fa270a1c5a0b52ab2f6784094e8022b57c8d8f2472deR193
You can either have it as `add_callback` and use fixtures or have `add_response` as it is now.

Reply:
omfg. I just merged this damn thing.

I'll open a new PR.